caml_z.c

```diff
@@ -954,7 +958,7 @@ CAMLprim value ml_z_to_bits(value arg)
   Z_ARG(arg);
   r = caml_alloc_string(size_arg * sizeof(mp_limb_t));
   Z_REFRESH(arg);
-  p = (unsigned char*) String_val(r);
+  p = Bytes_val(r);
```

Comment:
This change does not appear to follow the spirit of this comment:

https://github.com/ocaml/ocaml/issues/7594#issuecomment-473071206

Reply:
reverted